### PR TITLE
Improve support of image annotations

### PR DESCRIPTION
**Pull request by bencomp**
See the discussion at #27 for context. This PR:

- fixes the URIs for the `Annotation` and the `resource` (main open issue in #27)
- adds an endpoint for retrieving the (single) image-painting annotation using its URI
- adjust the endpoint for retrieving Canvas representations
- uses full-size Image API URIs so that image dimensions match the response and less processing is need for resizing on the fly 
- adds the IIIF Presentation API 2.1 `@context` to Canvas and Annotation responses
- adds and updates `@return` doc tags and descriptions to several functions
- adds module dependencies that were missing
- fixes `module_load_include` calls that used the wrong module

### CODE REVIEW

**Comment by bencomp** [comment]
Thanks to @ublrama, we were able to undo some of my naive 'fixes' and test this on the staging server.
I started from a branch that didn't include #39, so we recreated that commit.

All changes have been discussed, except for the extra module dependencies. It is likely that not listing them explicitly didn't cause issues, because they are installed on our servers anyway, or because they are transitive dependencies of already listed dependencies. Still, I feel it is good to have them here too, since this module directly includes and calls functions from `islandora_handle`, `islandora_paged_content` and `islandora_newspaper`.

**Comment by lucasvanschaik** [comment]
General comment: I see you are starting with docblock comments and have added `@return`. That is nice, but why stop there? How about some `@param` as well?

File: includes/manifest.inc [hunk 1]
```diff
@@ -55,18 +57,36 @@ function islandora_iiif_manifests_create_manifest_json(AbstractObject $object, $
...
+ * There should be exactly one such annotation in our Canvas, allowing us
+ * to build the Canvas and taking the image annotation from it.
+ * @return void
+ */
+function islandora_iiif_manifests_create_annotation_json(AbstractObject $object) {
+  
+  module_load_include('inc', 'islandora', 'includes/utilities');
+  
+  $output = islandora_iiif_manifests_build_canvas($object);
+  $annotation = $output['images'][0];
```
Comment (lucasvanschaik) [inline includes/manifest.inc]: Shouldn't you check existence first?

File: includes/utilities.inc [hunk 2]
```diff
@@ -94,7 +97,7 @@ function islandora_iiif_manifests_createDefaultCanvas($canvas_object) {
     ),
     'images' => array(
       array(
-        '@id' => $canvas_object['service_uri'],
+        '@id' => $canvas_object['id_uri'],
```
Comment (lucasvanschaik) [inline includes/utilities.inc:100]: Are you sure that key 'id_uri' exists?

File: includes/utilities.inc [hunk 3]
```diff
@@ -136,6 +140,7 @@ function islandora_iiif_manifests_get_object_canvas($object) {
   $canvas_object['thumbnail_uri'] = $image_info['thumbnail_uri'];
   $canvas_object['resource_uri'] = $image_info['resource_uri'];
   $canvas_object['service_uri'] = $image_info['service_uri'];
+  $canvas_object['id_uri'] = $image_info['id_uri'];
   $canvas_object['imageWidth'] = $image_info['imageWidth'];
   $canvas_object['imageHeight'] = $image_info['imageHeight'];
```
Comment (lucasvanschaik) [inline includes/utilities.inc]: Sure that all these exist?

File: includes/utilities.inc [hunk 4]
```diff
@@ -327,9 +342,9 @@ function islandora_iiif_manifests_get_image_info($object) {
   $handle = $handler->getFullHandle($object);
   
   $encoded_handle = urlencode('hdl:' . $handle);
```
Comment (lucasvanschaik) [inline includes/utilities.inc]: Not used

File: includes/utilities.inc [hunk 5]
```diff
@@ -327,9 +342,9 @@ function islandora_iiif_manifests_get_image_info($object) {
...
-  $handle_pieces = explode("/",$handle);
+  $handle_pieces = explode("/", $handle);
   
-  $new_handle = "hdl:".$handle_pieces[0]."%252F".$handle_pieces[1];
+  $new_handle = "hdl:" . $handle_pieces[0] . "%252F" . $handle_pieces[1];
```
Comment (lucasvanschaik) [inline includes/utilities.inc:343]: why not use urlencode? Twice? With comments on why this is happening?

File: includes/utilities.inc (hunk 2 above)
Comment (ublrama) [reply includes/utilities.inc:100]: Yes it does, here: https://github.com/bencomp/islandora_iiif_manifests/blob/1a4ee4a3e3c8482145a7a5351cb32718aaea5f3d/includes/utilities.inc#L358

File: includes/utilities.inc (hunk 3 above)
Comment (ublrama) [reply includes/utilities.inc]: Yes, though imageWidth and imageHeight depends on if the object has either a TECHMD or OBJ. The width and height are retrieved from either one of those datastreams. I assume that both datastreams are alway available. Or is that not the case?

File: includes/manifest.inc (hunk 1 above)
Comment (ublrama) [reply includes/manifest.inc]:
```
 // Check if an image is available
  if (isset($output['images'][0])) {
    $annotation = $output['images'][0];
    $annotation['@context'] = 'http://iiif.io/api/presentation/2/context.json';
  }
  else {
    $annotation = array(
      "data" => '{"error": "No manifest available"}',
      "status" => "404 Not Found"
    );
  }
  
  islandora_iiif_manifests_create_json_response('200 OK', $annotation);
```

**Pull request merged**